ravenml/data/write_dataset.py

```diff
@@ -0,0 +1,248 @@
+import os, shutil, time, json
+import contextlib2
+from pathlib import Path
+from object_detection.dataset_tools import tf_record_creation_util
+from random import shuffle
+from datetime import datetime
+from ravenml.data.interfaces import CreateInput
+from ravenml.utils.question import cli_spinner, cli_spinner_wrapper
+from ravenml.utils.config import get_config
+from ravenml.data.helpers import filter_sets, load_metadata, ingest_metadata
+from ravenml.utils.io_utils import download_data_from_s3, copy_data_locally
+
+
+class DatasetWriter(object):
+
+    def __init__(self, create: CreateInput, **kwargs):
+        self.associated_files = kwargs['associated_files']
+        self.related_data_prefixes = kwargs['related_data_prefixes']
+        
+        metadata = create.metadata
+        self.num_folds = metadata['kfolds'] if metadata.get('kfolds') else 5 # Never actually used
+        self.test_percent = metadata['test_percent'] if metadata.get('test_percent') else .2
+        self.label_to_int_dict = {}
+        self.image_ids = None
+        self.filter_metadata = None
+        self.temp_dir = create.plugin_metadata['temp_dir_path']
+        self.dataset_path = create.dataset_path
+        self.dataset_name = metadata['dataset_name']
+        self.created_by = metadata['created_by']
+        self.comments = metadata['comments']
+        self.plugin_name = create.plugin_metadata['architecture']
+        self.imagesets_used = metadata['imagesets_used']
+        self.filter = metadata['filter']
+        self.cache = create.plugin_cache
+    
+    def construct_all(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def export_as_TFExample(self, *args, **kwargs):
+        raise NotImplementedError
+    
+    def write_additional_files(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def filter_and_load(self, metadata_prefix):
+        bucketConfig = get_config()
+        image_bucket_name = bucketConfig.get('image_bucket_name')
+
+        cli_spinner("Loading metadata...", ingest_metadata, self.imagesets_used, image_bucket_name, self.cache)
+
+        tags_df = load_metadata(metadata_prefix, self.cache)
+        filter_metadata = {"groups": []}
+
+        if self.filter:
+            image_ids = filter_sets()
+        else: 
+            image_ids = tags_df.index.tolist()
+                
+        # condition function for S3 download and local copying
+        def need_file(filename):
+            if len(filename) > 0:
+                image_id = filename[filename.index('_')+1:filename.index('.')]
```

Comment:
Running into an issue where files in the imageset that do not fit the `name_####.extension` naming scheme break the plugin. Would be great if the filename could be disregarded if it does not fit the naming scheme.

Reply:
I'm looking into it now

Reply:
Can you give me the use case for when the plugin is breaking? The need_file function is from jigsaw, but I think it is intended to return True if the filename matches the format you described and not return otherwise. I see it is being used by the copy_data_locally and the download_from_s3 where the files being looked at must get True from the function to be acted upon. I have changed the function to now explicitly return False if the filename does not match the format, but am not sure whether this resolves the issue.

Reply:
it looks like a recent commit fixes this, can we resolve?